woocommerce/woocommerce-gateway-stripe
# Improve branded Google Pay button user agent detection

The current implementation for Chrome detection is matching some Chromium-based browsers.
Achieving 100% coverage of existing browsers and maintaining the checks is a huge effort, but we should do it for the most used ones. This commit adds checks to prevent the GPay button from rendering in Edge, Opera, and Brave.

Fixes #1214.

#### Changes proposed in this Pull Request:

- Check if the browser is not Edge, Opera, or Brave before displaying the Google Pay styled button

#### Testing instructions:

1. Have "Branded" set as the "Payment Request Button Type"
2. Open Microsoft Edge / Brave / Opera
3. Allow save and fill payment methods
    3.1 opera://settings/payments
    3.2 brave://settings/payments
    3.3 edge://settings/payments
4. Add any payment card, such as the test 4242 one
5. Load a product page, the G Pay branding should not be applied
6. Test it with Chrome, the G Pay branding should be applied
